src/core/unit.c

```diff
@@ -1742,6 +1745,23 @@ int unit_load(Unit *u) {
                 unit_invalidate_cgroup_members_masks(u);
         }
 
+        /* For top-level cgroups, if not set during loading, restrict allowed CPUs based on manager value */
+        if(u->manager->defaults.allowed_cpus.allocated > 0 && UNIT_HAS_CGROUP_CONTEXT(u)) {
+                cc = unit_get_cgroup_context(u);
+                if(!cc)
+                        goto fail;
+
+                if(cc->cpuset_cpus.allocated == 0) {
```

Comment:
please follow coding style, i.e. space between `if` and `(`, here and everywhere else.

Reply:
also, please abstract this in a small helper cpuset_isempty() or so